### PR TITLE
[Tooltip] Tweak animation duration and box-shadow

### PR DESCRIPTION
**Pull request by mrcthms**
### WHY are these changes introduced?

After liaising with @dGoligorsky we wanted to tweak some of the primitive values for the Tooltip, namely the animation duration and the box-shadow for the overlay. The animation felt too slow, and the shadow meant that Tooltips being displayed over white backgrounds were quite hard to distinguish.

We have sped up the animation duration from 200ms to 50ms, and added an extra shadow on top of the `--p-shadow-popover` for the Tooltip to ensure the top edge is visible.

Spin URL: https://admin.web.tooltip-tweak-demo.marc-thomas.eu.spin.dev/store/shop1/products/1

### 🎩 checklist

- [x] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [x] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [x] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [x] Updated the component's `README.md` with documentation changes
- [x] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

**Comment by mrcthms** [comment]
@Shopify/polaris-eng Can I get a review of this little one please? Thx!

File: polaris-react/src/components/Tooltip/components/TooltipOverlay/TooltipOverlay.scss [hunk 1]
```diff
@@ -7,7 +7,8 @@
   // stylelint-disable-next-line -- generated by polaris-migrator DO NOT COPY
   margin: var(--pc-tooltip-overlay-offset) var(--p-space-4) var(--p-space-4);
   opacity: 1;
-  box-shadow: var(--p-shadow-popover);
+  // stylelint-disable-next-line -- hardcoding an extra shadow for the Tooltip to enable it to be more visible against white backgrounds
+  box-shadow: var(--p-shadow-popover), 0 0 20px 8px rgba(23, 24, 24, 0.02);
```
Comment (kyledurand) [inline polaris-react/src/components/Tooltip/components/TooltipOverlay/TooltipOverlay.scss]:
```suggestion
  box-shadow: var(--p-shadow-popover), var(--p-shadow-deep);
```
Can we do something like this? It seems to only increase in contrast which should be a good thing and we stick with tokens

**Review by kyledurand — APPROVED** [review]
Changes look good, just the one comment about using a secondary token vs adding a snowflake

File: polaris-react/src/components/Tooltip/components/TooltipOverlay/TooltipOverlay.scss (hunk 1 above)
Comment (lgriffee) [reply polaris-react/src/components/Tooltip/components/TooltipOverlay/TooltipOverlay.scss]: Also want to add in v11 we will be switching from our current [`depth` tokens](https://github.com/Shopify/polaris/blob/main/polaris-tokens/src/token-groups/depth.ts) to these new [`shadow` tokens](https://github.com/Shopify/polaris/blob/main/polaris-tokens/src/token-groups/shadow.ts) (we have [a migration](https://github.com/Shopify/polaris/pull/8272/files#diff-39620b2a876df203007dbee19f2d949f335ec9403499f63a3c0eb1dfbb2afeb5R7-R18) that maps all current values to new ones but if you want to be able to hand pick the values yourself I'd recommend using the new shadow tokens here instead).

Comment (dGoligorsky) [reply polaris-react/src/components/Tooltip/components/TooltipOverlay/TooltipOverlay.scss]: Looking at the upcoming shadow tokens, it seems that `--p-shadow-lg` will do the trick

Comment (mrcthms) [reply polaris-react/src/components/Tooltip/components/TooltipOverlay/TooltipOverlay.scss]: Looks like `--p-shadow-lg` doesn't exist currently, are we okay to hardcode it for now with a comment to update it once we move to v11.0?

Comment (alex-page) [reply polaris-react/src/components/Tooltip/components/TooltipOverlay/TooltipOverlay.scss]: It should exist. You might need to run a local build and get latest from `main` if it isn't there. https://github.com/Shopify/polaris/blob/764f17d86c30c9f30bec6717c09f07b2e04e23f8/polaris-tokens/src/token-groups/shadow.ts#L46-L49

**Comment by mrcthms** [comment]
Is this okay to merge now @kyledurand ?

**Comment by kyledurand** [comment]
🚢 !

**Pull request merged**